packages/editor/src/plugins/rows/components/plugin-menu-modal.tsx

```diff
@@ -0,0 +1,224 @@
+import { EditorPluginType } from '@editor/types/editor-plugin-type'
+import { useEditorStrings } from '@serlo/frontend/src/contexts/logged-in-data-context'
+import React, { useContext, useEffect, useMemo, useRef, useState } from 'react'
+import { Key } from 'ts-key-enum'
+
+import { PluginMenuItem } from './plugin-menu-item'
+import type { PluginMenuItemType } from '../contexts/plugin-menu/types'
+import { usePluginMenuKeyboardHandler } from '../hooks/use-plugin-menu-keyboard-handler'
+import {
+  createOption,
+  filterOptions,
+  isInteractivePluginType,
+} from '../utils/plugin-menu'
+import { ModalWithCloseButton } from '@/components/modal-with-close-button'
+
+interface PluginMenuModalProps {
+  onInsertPlugin: (pluginType: EditorPluginType) => void
+}
+
+export function PluginMenuModal({ onInsertPlugin }: PluginMenuModalProps) {
+  const editorStrings = useEditorStrings()
+  const pluginsStrings = editorStrings.plugins
+
+  const { pluginMenuState, pluginMenuDispatch } = useContext(PluginMenuContext)
+
+  const [searchString, setSearchString] = useState('')
+  const [focusedItemIndex, setFocusedItemIndex] = useState<number | null>(null)
+  const itemRefs = useRef<(HTMLButtonElement | null)[]>([])
+
+  const searchInputRef = useRef<HTMLInputElement | null>(null)
+
+  const allWithData = editorPlugins.getAllWithData()
+  const allowedPlugins = useMemo(() => {
+    const allVisible = allWithData
+      .filter(
+        ({ type, visibleInSuggestions }) =>
+          visibleInSuggestions ||
+          isInteractivePluginType(type as EditorPluginType)
+      )
+      .map(({ type }) => type)
+
+    const allowedByContext = pluginMenuState.allowedChildPlugins ?? allVisible
+    // Filter out plugins which can't be nested inside of the current plugin or ancestor plugins
+    const typesOfAncestors = selectAncestorPluginTypes(store.getState(), ROOT)
+
+    return typesOfAncestors
+      ? allowedByContext.filter((plugin) =>
+          checkIsAllowedNesting(plugin, typesOfAncestors)
+        )
+      : allowedByContext
+  }, [allWithData, pluginMenuState.allowedChildPlugins])
+
+  const { basicOptions, interactiveOptions, isEmpty } = useMemo(() => {
+    const allOptions = allowedPlugins.map((type) => {
+      return createOption(type as EditorPluginType, pluginsStrings)
+    })
+    const allowed = filterOptions(allOptions, searchString)
+
+    const basicOptions = allowed.filter(
+      (option) => !isInteractivePluginType(option.pluginType)
+    )
+
+    const interactiveOptions = allowed.filter((option) =>
+      isInteractivePluginType(option.pluginType)
+    )
+
+    return {
+      basicOptions,
+      interactiveOptions,
+      isEmpty: basicOptions.length === 0 && interactiveOptions.length === 0,
+    }
+  }, [allowedPlugins, pluginsStrings, searchString])
+
+  const handleItemFocus = (index: number) => {
+    setFocusedItemIndex(index)
+  }
+
+  const handleItemBlur = () => {
+    setFocusedItemIndex(-1)
```

Comment:
you use null everywhere else, is that `-1` on purpose?

Reply:
(changed in https://github.com/serlo/frontend/pull/4021)

Reply:
>you use null everywhere else, is that -1 on purpose?

I introduced the `null` and overlooked this one instance. My bad.